src/app/hero.service.ts

```diff
@@ -1,23 +1,101 @@
 import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { Observable, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
 import { Hero } from './hero';
-import { HEROES } from './mock-heroes';
 import { MessageService } from './message.service';
 
+
 @Injectable({ providedIn: 'root' })
 export class HeroService {
 
-  constructor(private messageService: MessageService) { }
+  private heroesUrl = 'api/heroes';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/infinity-arc/Kobus-tour-of-heroes/pullRequest?prid=7838883)
```suggestion
  private heroesUrl = "api/heroes";
```